trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -5,44 +5,58 @@ class SmartBoard(models.Model):
     _description = "SmartBoard Connector"
 
     @api.model
-    def create_project(self, sb_lead_id, project_template_id):
+    def create_project(self, sb_lead_id, project_template_id=None):
         try:
-            if sb_lead_id and project_template_id:
-                partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
-                lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
-                project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
-                partner_id = []
-
-                if not partner_exist:
-                    # Create new partner
-                    new_id = self.env['res.partner'].create({'name': 'SBLead-' + sb_lead_id, 'sb_lead_id': sb_lead_id, "sync_status" : '1'})
-                    partner_id.append(new_id)
-
-                if not lead_exist:
-                    # Create new lead
-                    lead_id = self.env['crm.lead'].create({'name': 'SBLead-' + sb_lead_id, 'sb_lead_id': sb_lead_id})
-                    # Convert lead to opportunity
-                    if partner_id:
-                        customer = self.env['res.partner'].browse(partner_id[0])
+            if sb_lead_id:
+                if type(sb_lead_id) == int:
+                    partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                    lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                    project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+
+                    if not partner_exist:
+                        # Create new partner
+                        partner_new = self.env['res.partner'].create({'name': 'SBLead-' + str(sb_lead_id), 'sb_lead_id': sb_lead_id, "sync_status" : '1'})
+
+                    if not lead_exist:
+                        # Create new lead
+                        lead_new = self.env['crm.lead'].create({'name': 'SBLead-' + str(sb_lead_id), 'sb_lead_id': sb_lead_id})
+                        # Convert lead to opportunity
+                        if partner_exist:
+                            vals = lead_new[0]._convert_opportunity_data(partner_exist, team_id=False)
```

Comment:
Is lead_new a single record or a list of records?

Reply:
resolved in v4